### PR TITLE
Feature/query cxx praveen

### PR DESCRIPTION
**Pull request by PraveenKumar-NOAA**
## Description

This PR updates adpupa prepbufr yaml for the group_by field and adds a separate adpupa prepbufr yaml for only GSI mnemonics.

The following files will be added in this PR: 

- adpupa_prepbufr_group_by_v2.yaml
- adpupa_prepbufr_group_by_v2.nc
- adpupa_gsiprepbufr_group_by_v2.yaml
- adpupa_gsiprepbufr_group_by_v2.nc

### Issue(s) addressed
#514
#515 

## Acceptance Criteria (Definition of Done)
Working plus test data validation. 

## Dependencies
None

## Impact
None

### CODE REVIEW

**Comment by emilyhcliu** [comment]
@PraveenKumar-NOAA 
Could you change the name of your YAMLS and test files as the following:

```
adpupa_prepbufr_group_by_v2.yaml ---> bufr_ncep_prepbufr_adpupa.yaml
adpupa_prepbufr_group_by_v2.nc ---> bufr_ncep_prepbufr_adpupa.nc
adpupa_gsiprepbufr_group_by_v2.yaml ---> bufr_ncep_prepbufr_adpupa_gsi.yaml
adpupa_gsiprepbufr_group_by_v2.nc ---> bufr_ncep_prepbufr_adpupa_gsi.nc
```

**Comment by emilyhcliu** [comment]
Note: 
(1) The stationIdentifier should have the dimension of nlocs_prepbufrDataLvlCat.  (each observation value should have an station ID)
(2) The stationElevation should be in MetaData group (not ObsValue)

**Comment by emilyhcliu** [comment]
The Enumeration data type should be unitness.  Therefore, please change the following:
```
units: "ENUMETAION"  to  units: ""

```

**Comment by emilyhcliu** [comment]
obsTimeMinusCycleTime should be MetaData.

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+#        - name: tevent_Dim
+#          path: "*/PRSLEVEL/T___INFO/T__EVENT"
+#        - name: zevent_Dim
+#          path: "*/PRSLEVEL/Z___INFO/Z__EVENT"
+#        - name: wevent_Dim
+#          path: "*/PRSLEVEL/W___INFO/W__EVENT"
+#
+#     dimensions:
+#       - name: "nlocs" 
+#         size: variables/latitude.nrows
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:164]: Remove these three lines.  They are obsolete.

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+
+      variables:
+
+        - name: "stationIdentification@MetaData"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:169]: Let's do this:
-name: "MetaData/stationIdentification"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/stationElevation
+          longName: "Height of Station"
+          units: "Meter"
+
+        - name: "prepbufrReportType@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/prepbufrReportType
+          longName: "Prepbufr Report Type"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:205]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "dumpReportType@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/dumpReportType
+          longName: "Data Dump Report Type"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:211]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "prepbufrDataLvlCat@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/prepbufrDataLvlCat
+          longName: "Prepbufr Data Level Category"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:217]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "methodofTemperatureSeaSurfaceMeasurement@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/methodofTemperatureSeaSurfaceMeasurement
+          longName: "Method of Sea Surface Measurement"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:223]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "presentWeather@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/presentWeather
+          longName: "Present Weather"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:229]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 8]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "verticalSignificanceSurfaceObservations@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/verticalSignificanceSurfaceObservations
+          longName: "Vertical Significance (Surface Observations)"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:235]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 9]
```diff
@@ -0,0 +1,406 @@
...
+
+        - name: "cloudAmount@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/cloudAmount
+          longName: "Cloud Amount"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:241]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 10]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/cloudCoverTotal
+          longName: "Cloud Cover"
+          units: "Percent"
+
+        - name: "heightAboveSurfaceofBaseofLowestCloud@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/heightAboveSurfaceofBaseofLowestCloud
+          longName: "Height above Surface of Base of Lowest Cloud"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:259]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 11]
```diff
@@ -0,0 +1,406 @@
...
+          source: variables/pressure
+          longName: "Pressure"
+          units: "Pa"
+#         range: [20000, 110000]
+
+        - name: "pressureQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/pressureQM
+          longName: "Pressure QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:279]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 12]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/specificHumidity
+          longName: "Specific Humidity"
+          units: "Kilogram Kilogram-1"
+
+        - name: "specificHumidityQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/specificHumidityQM
+          longName: "Specific Humidity QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:291]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 13]
```diff
@@ -0,0 +1,406 @@
...
+          source: variables/airTemperature
+          longName: "Temperature"
+          units: "Kelvin"
+#         range: [193, 325]
+
+        - name: "airTemperatureQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/airTemperatureQM
+          longName: "Temperature QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:304]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 14]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/heightOfObservation
+          longName: "Height of Observation"
+          units: "Meter"
+
+        - name: "heightQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/heightQM
+          longName: "Height QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:316]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 15]
```diff
@@ -0,0 +1,406 @@
...
+          source: variables/windNorthward
+          longName: "Northward Wind"
+          units: "Meter Second-1"
+#         range: [-50, 50]
+
+        - name: "windQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/windQM
+          longName: "U, V-Component of Wind QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:336]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 16]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/seaSurfaceTemperature
+          longName: "Sea Surface Temperature"
+          units: "Kelvin"
+
+        - name: "seaSurfaceTemperatureQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/seaSurfaceTemperatureQM
+          longName: "Sea Surface Temperature QM"
+          units: "ENUMERATION"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:392]: Change units: "ENUMERATION: to units: ""

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 17]
```diff
@@ -0,0 +1,406 @@
...
+          range: [0, 360]
+
+        - name: "latitude@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/latitude
+          longName: "Latitude"
+          units: "degrees_north"
+          range: [-90, 90]
+
+        - name: "obsTimeMinusCycleTime@ObsValue"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:189]: This one should be MetaData/obsTimeMinusCycleTime

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 18]
```diff
@@ -0,0 +1,406 @@
...
+          longName: "Data Dump Report Type"
+          units: "ENUMERATION"
+
+        - name: "prepbufrDataLvlCat@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/prepbufrDataLvlCat
+          longName: "Prepbufr Data Level Category"
+          units: "ENUMERATION"
+
+        - name: "methodofTemperatureSeaSurfaceMeasurement@ObsValue"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:219]: This one should belong to MetaData group.

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 19]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/methodofTemperatureSeaSurfaceMeasurement
+          longName: "Method of Sea Surface Measurement"
+          units: "ENUMERATION"
+
+        - name: "presentWeather@MetaData"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:225]: I think this one should be in ObsValue group.

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 20]
```diff
@@ -0,0 +1,406 @@
...
+#         range: [193, 325]
+
+        - name: "pressure@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/pressure
+          longName: "Pressure"
+          units: "Pa"
+#         range: [20000, 110000]
+
+        - name: "pressureQM@QualityMarker"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:275]: Change to the following:
-name: "QualityMarker/pressure"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 21]
```diff
@@ -0,0 +1,406 @@
...
+          longName: "Pressure QM"
+          units: "ENUMERATION"
+
+        - name: "specificHumidity@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/specificHumidity
+          longName: "Specific Humidity"
+          units: "Kilogram Kilogram-1"
+
+        - name: "specificHumidityQM@QualityMarker"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:287]: Change to the following:
-name: "QualityMarker/specificHumidity"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 22]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/pressure
+          longName: "Pressure"
+          units: "Pa"
+#         range: [20000, 110000]
+
+        - name: "pressureQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/pressureQM
+          longName: "Pressure QM"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:278]: Change "Pressure QM" to "Pressure Quality Marker"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 23]
```diff
@@ -0,0 +1,406 @@
...
+        - name: "specificHumidity@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/specificHumidity
+          longName: "Specific Humidity"
+          units: "Kilogram Kilogram-1"
+
+        - name: "specificHumidityQM@QualityMarker"
+          coordinates: "longitude latitude"
+          source: variables/specificHumidityQM
+          longName: "Specific Humidity QM"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:290]: Change "Specific Humidity QM" to "Specific Humidity Quality Marker"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 24]
```diff
@@ -0,0 +1,406 @@
...
+          units: "ENUMERATION"
+
+        - name: "airTemperature@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/airTemperature
+          longName: "Temperature"
+          units: "Kelvin"
+#         range: [193, 325]
+
+        - name: "airTemperatureQM@QualityMarker"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:300]: Change to "QualityMarker/airTemperature"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 25]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/airTemperatureQM
+          longName: "Temperature QM"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:303]: Change "Temperature QM" to "Temperature Quality Marker"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 26]
```diff
@@ -0,0 +1,406 @@
...
+#         range: [-50, 50]
+
+        - name: "windNorthward@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/windNorthward
+          longName: "Northward Wind"
+          units: "Meter Second-1"
+#         range: [-50, 50]
+
+        - name: "windQM@QualityMarker"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:332]: Change to "QualityMarker/wind"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 27]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/windQM
+          longName: "U, V-Component of Wind QM"
+          units: "ENUMERATION"
+
+        - name: "pressureError@ObsError"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:338]: Change to "ObsError/pressure"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 28]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/pressureError
+          longName: "Pressure Error"
+          units: "Pa"
+
+        - name: "relativeHumidityError@ObsError"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:344]: Change to "ObsError/relativeHumidity"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 29]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/relativeHumidityError
+          longName: "Relative Humidity Error"
+          units: "Percent divided by 10"
+
+        - name: "airTemperatureError@ObsError"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:350]: Change to "ObsError/airTemperature"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 30]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/airTemperatureError
+          longName: "Temperature Error"
+          units: "Kelvin"
+
+        - name: "windError@ObsError"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:356]: Change to "ObsError/wind"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 31]
```diff
@@ -0,0 +1,406 @@
...
+          range: [0, 360]
+
+        - name: "latProfileLevel@MetaData"
+          coordinates: "longitude latitude"
+          source: variables/latProfileLevel
+          longName: "Latitude Profile Level"
+          units: "degrees_north"
+          range: [-90, 90]
+
+        - name: "timeCycleProfileLevel@ObsValue"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:376]: This one should be in MetaData group.

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 32]
```diff
@@ -0,0 +1,406 @@
...
+          longName: "Time Cycle Profile Level"
+          units: "Hours"
+
+        - name: "seaSurfaceTemperature@ObsValue"
+          coordinates: "longitude latitude"
+          source: variables/seaSurfaceTemperature
+          longName: "Sea Surface Temperature"
+          units: "Kelvin"
+
+        - name: "seaSurfaceTemperatureQM@QualityMarker"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:388]: Change to "QualityMarker/seaSurfaceTemperature"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 33]
```diff
@@ -0,0 +1,406 @@
...
+          coordinates: "longitude latitude"
+          source: variables/seaSurfaceTemperatureQM
+          longName: "Sea Surface Temperature QM"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:391]: Change to "Sea Surface Temperature Quality Marker"

File: test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml [hunk 34]
```diff
@@ -0,0 +1,406 @@
...
+          units: "ENUMERATION"
+
+        - name: "seaSurfaceTemperatureError@ObsError"
```
Comment (emilyhcliu) [inline test/testinput/adpupa_gsiprepbufr_group_by_v2.yaml:394]: Change to "ObsError/seaSurfaceTemperature"

**Comment by emilyhcliu** [comment]
@PraveenKumar-NOAA You have two YAML files:
adpupa_prepbufr_group_by_v2.yaml
adpupa_gsiprepbufr_group_by_v2.yaml

These two YAMLs are basically the same except the first one has two additional Mnemonics:TVO and CLTP.
I suggest you add these TVO and CLTP to adpupa_gsiprepbufr_group_by_v2.yaml and remove the other YAML.

And, then rename adpupa_gsiprepbufr_group_by_v2.yaml to bufr_ncep_prepbufr_adpupa.yaml

What do you think?

**Comment by PraveenKumar-NOAA** [comment]
> @PraveenKumar-NOAA You have two YAML files: adpupa_prepbufr_group_by_v2.yaml adpupa_gsiprepbufr_group_by_v2.yaml
> 
> These two YAMLs are basically the same except the first one has two additional Mnemonics:TVO and CLTP. I suggest you add these TVO and CLTP to adpupa_gsiprepbufr_group_by_v2.yaml and remove the other YAML.
> 
> And, then rename adpupa_gsiprepbufr_group_by_v2.yaml to bufr_ncep_prepbufr_adpupa.yaml
> 
> What do you think?

Sounds great. I added TVO and CLTP to my yaml and made changes accordingly.

File: test/testinput/bufr_ncep_prepbufr_adpupa.yaml [hunk 35]
```diff
@@ -0,0 +1,425 @@
...
+#          path: "*/PRSLEVEL/P___INFO/P__EVENT"
+#        - name: qevent_Dim
+#          path: "*/PRSLEVEL/Q___INFO/Q__EVENT"
+#        - name: tevent_Dim
+#          path: "*/PRSLEVEL/T___INFO/T__EVENT"
+#        - name: zevent_Dim
+#          path: "*/PRSLEVEL/Z___INFO/Z__EVENT"
+#        - name: wevent_Dim
+#          path: "*/PRSLEVEL/W___INFO/W__EVENT"
+#
```
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_prepbufr_adpupa.yaml:170]: We can use these dimensions.  @rhoneyager will add this to ObsSpace.yaml when we send this yaml for PR (to be merged to develop) in the future.

File: test/testinput/bufr_ncep_prepbufr_adpupa.yaml [hunk 36]
```diff
@@ -0,0 +1,425 @@
...
+
+
+    ioda:
+      backend: netcdf 
+      obsdataout: "./testrun/bufr_ncep_prepbufr_adpupa.nc"
+
+
+#      dimensions:
+##       - name: prslevel_Dim -> Level
+#        - name: Level
```
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_prepbufr_adpupa.yaml:156]: Level is good here.  You can remove the ## line.

File: test/testinput/bufr_ncep_prepbufr_adpupa.yaml [hunk 37]
```diff
@@ -0,0 +1,425 @@
...
+#          path: "*/PRSLEVEL"
+#        - name: cloudseq_Dim
+#          path: "*/CLOUDSEQ"
```
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_prepbufr_adpupa.yaml:159]: The dimension related to cloud observations from cldseq requires a bit more thinking.  
Let's just leave it as cloudseq_Dim for now.  But, we will have to replace it with a better name.

I have been trying to find the physical meaning of this dimension.  The [WMO manual 306](https://library.wmo.int/doc_num.php?explnum_id=10235) has the information.  Search " cloud amount"  in  the linked PDF.  However, the information in the manual does not mention the type of information sotred in cloudseq_Dim.  These cloud observations are used by RTMA/regional group.  I will reach out to the RTMA group for more information.

File: test/testinput/bufr_ncep_prepbufr_adpupa.yaml [hunk 38]
```diff
@@ -0,0 +1,425 @@
...
+          seaSurfaceTemperatureError:
+            query: "*/SST_INFO/SSTBACKG/SSTOE"
+
+          depthBelowSeaSurface:
+            query: "*/SST_INFO/DBSS_SEQ/DBSS"
+
+
+    ioda:
+      backend: netcdf 
+      obsdataout: "./testrun/bufr_ncep_prepbufr_adpupa.nc"
```
Comment (emilyhcliu) [inline test/testinput/bufr_ncep_prepbufr_adpupa.yaml:151]: Do we want to unified the output filename as the following:
ioda_ncep_prepbufr_adpupa.nc ?  
What do you think?

**Comment by emilyhcliu** [comment]
Validation Pots for Temperature (Sensible + Virtual): IODA vs. GSI
gsi_ncep_prepbufr_adpupa_temperature
ioda_ncep_prepbufr_adpupa_temperature

**Comment by emilyhcliu** [comment]
Validation plots for Specific Humidity: IODA vs. GS
gsi_ncep_prepbufr_adpupa_sphum
ioda_ncep_prepbufr_adpupa_sphum
I

**Comment by emilyhcliu** [comment]
Validation plots for Winds: GSI vs. IODA
gsi_ncep_prepbufr_adpupa_uwind
gsi_ncep_prepbufr_adpupa_vwind
ioda_ncep_prepbufr_adpupa_uwind
ioda_ncep_prepbufr_adpupa_vwind

**Comment by emilyhcliu** [comment]
Validation plots for surface pressure: IODA vs. GSI
gsi_ncep_prepbufr_adpupa_psfc
ioda_ncep_prepbufr_adpupa_psfc

**Comment by emilyhcliu** [comment]
@PraveenKumar-NOAA Please also change the following three BUFR variables to be used with "group_by: prepbufrDataLvlCat"

```
          stationElevation:
            query: "*/ELV"
            group_by: prepbufrDataLvlCat

          prepbufrReportType:
            query: "*/TYP"
             group_by: prepbufrDataLvlCat

          dumpReportType:
            query: "*/T29"
            group_by: prepbufrDataLvlCat
```

**Comment by PraveenKumar-NOAA** [comment]
> @PraveenKumar-NOAA Please also change the following three BUFR variables to be used with "group_by: prepbufrDataLvlCat"
> 
> ```
>           stationElevation:
>             query: "*/ELV"
>             group_by: prepbufrDataLvlCat
> 
>           prepbufrReportType:
>             query: "*/TYP"
>              group_by: prepbufrDataLvlCat
> 
>           dumpReportType:
>             query: "*/T29"
>             group_by: prepbufrDataLvlCat
> ```

Done

**Review by emilyhcliu — APPROVED** [review]

**Comment by rmclaren** [comment]
Is this one ready to be merged?

**Review by rmclaren — APPROVED** [review]

**Pull request merged**